server/api/contest_handler.py

```diff
@@ -13,12 +13,29 @@
 @contest_handler.route('', methods=['GET'])
 def show_all():
     user_id = request.args.get('user_id')
+    completed = request.args.get('completed')
+    current_time = datetime.utcnow()
 
     # Retrieve all contests for specific user
     if user_id != None:
+        if completed == "True":
+            contests = Contest.query.filter(Contest.deadline <= current_time).filter_by(user_id=user_id).all()
+            return jsonify([contest.to_dict() for contest in contests])
+        elif completed == "False":
+            contests = Contest.query.filter(Contest.deadline > current_time).filter_by(user_id=user_id).all()
+            return jsonify([contest.to_dict() for contest in contests])
+
         contests = Contest.query.filter_by(user_id=user_id)
         return jsonify([contest.to_dict() for contest in contests])
 
+    # No specific user
+    if completed == "True":
+        contests = Contest.query.filter(Contest.deadline <= current_time).all()
+        return jsonify([contest.to_dict() for contest in contests])
+    elif completed == "False":
+        contests = Contest.query.filter(Contest.deadline > current_time).all()
+        return jsonify([contest.to_dict() for contest in contests])
+
```

Comment:
i'm sure there's a much more elegant way to do this, but this is functional for now